kernel/kernel-api/src/test/scala/io/delta/kernel/internal/PageTokenSuite.scala

```diff
@@ -0,0 +1,104 @@
+package io.delta.kernel.internal
+
+import java.util
+import java.util.{HashMap, Map}
+import java.util.Optional
+
+import scala.collection.JavaConverters._
+
+import io.delta.kernel.data.Row
+import io.delta.kernel.internal.annotation.VisibleForTesting
+import io.delta.kernel.internal.data.GenericRow
+import io.delta.kernel.internal.replay.PageToken
+import io.delta.kernel.test.MockFileSystemClientUtils
+import io.delta.kernel.types._
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class PageTokenSuite extends AnyFunSuite with MockFileSystemClientUtils {
+
+  private val TEST_FILE_NAME = "test_file.json"
+  private val TEST_ROW_INDEX = 42L
+  private val TEST_SIDECAR_INDEX = Optional.of(java.lang.Long.valueOf(5L))
+  private val TEST_KERNEL_VERSION = "4.0.0"
+  private val TEST_TABLE_PATH = "/path/to/table"
+  private val TEST_TABLE_VERSION = 5L
+  private val TEST_PREDICATE_HASH = 123L
+  private val TEST_LOG_SEGMENT_HASH = 456L
+
+  private val expectedPageToken = new PageToken(
+    TEST_FILE_NAME,
+    TEST_ROW_INDEX,
+    TEST_SIDECAR_INDEX,
+    TEST_KERNEL_VERSION,
+    TEST_TABLE_PATH,
+    TEST_TABLE_VERSION,
+    TEST_PREDICATE_HASH,
+    TEST_LOG_SEGMENT_HASH)
+
+  private val rowData: Map[Integer, Object] = new HashMap()
+  rowData.put(0, TEST_FILE_NAME)
+  rowData.put(1, TEST_ROW_INDEX.asInstanceOf[Object])
+  rowData.put(2, TEST_SIDECAR_INDEX.orElse(null))
+  rowData.put(3, TEST_KERNEL_VERSION)
+  rowData.put(4, TEST_TABLE_PATH)
+  rowData.put(5, TEST_TABLE_VERSION.asInstanceOf[Object])
+  rowData.put(6, TEST_PREDICATE_HASH.asInstanceOf[Object])
+  rowData.put(7, TEST_LOG_SEGMENT_HASH.asInstanceOf[Object])
+
+  val expectedRow = new GenericRow(PageToken.PAGE_TOKEN_SCHEMA, rowData)
+
+  test("Test PageToken.fromRow with valid data") {
+    val pageToken = PageToken.fromRow(expectedRow)
+    assert(pageToken.equals(expectedPageToken))
+  }
+
+  test("Test PageToken.toRow with valid data") {
+    val row = expectedPageToken.toRow
+    assert(row.getSchema.equals(PageToken.PAGE_TOKEN_SCHEMA))
+
+    assert(row.getString(0) == TEST_FILE_NAME)
+    assert(row.getLong(1) == TEST_ROW_INDEX)
+    assert(Optional.of(if (row.isNullAt(2)) null else row.getLong(2)) == TEST_SIDECAR_INDEX)
+    assert(row.getString(3) == TEST_KERNEL_VERSION)
+    assert(row.getString(4) == TEST_TABLE_PATH)
+    assert(row.getLong(5) == TEST_TABLE_VERSION)
+    assert(row.getLong(6) == TEST_PREDICATE_HASH)
+    assert(row.getLong(7) == TEST_LOG_SEGMENT_HASH)
+  }
+
+  test("E2E: PageToken round-trip: toRow -> fromRow") {
+    val row = expectedPageToken.toRow
+    val reconstructedPageToken = PageToken.fromRow(row)
+    assert(reconstructedPageToken.equals(expectedPageToken))
+  }
+
+  test("PageToken.fromRow throws exception when input row has invalid schema") {
```

Comment:
can we test:
- wrong field name
- wrong data type
- also explicitly test the sidecar-can-be-null case